### ISSUE TITLE
Implementation of full WPS 2.0.0 standard

### ISSUE THREAD

**Issue by jachym** [geopython/pywps#74]
Implementation of http://opengeospatial.org/standards/wps 2.0.0

**Comment by tomkralidis** [comment]
@jachym is this related to #56 in any way?

**Comment by jachym** [comment]
#56 is kind of pre-step to some of 2.0.0 features, yes, it's related



út 16. 2. 2016 v 0:59 odesílatel Tom Kralidis <notifications@github.com>
napsal:

**Comment by ldesousa** [comment]
The coming ReST specification for WPS 2.0 will have an impact on the implementation.

**Comment by cehbrecht** [comment]
There is processing interface which already introduces a `cancel` method:
https://github.com/geopython/pywps/blob/91da95ef846e4d9bce7cfd8016af467d2acfd1cb/pywps/processing/basic.py#L21 

We are having several implementations of the processing interface:
* multiprocessing (no `cancel` )
* scheduler
* docker: in the making #61.
* celery? (#237)

**Comment by cehbrecht** [comment]
@jachym @ldesousa 
can we use the mulitprocessing `terminate` or `kill` method to implement the WPS 2.0.0 *cancel* feature?

https://docs.python.org/3.7/library/multiprocessing.html#multiprocessing.Process.terminate

**Comment by ldesousa** [comment]
> @jachym @ldesousa
> can we use the mulitprocessing `terminate` or `kill` method to implement the WPS 2.0.0 _cancel_ feature?
> 
> https://docs.python.org/3.7/library/multiprocessing.html#multiprocessing.Process.terminate

Yes, that was the idea. The problem regarding WPS 2.0 is with the _Pause_ feature, there is nothing similar in the Multiprocessing library. 

Ideally, only the `terminate` would be used, `kill` might result in nasty things. But this needs to be tested.

**Comment by fmigneault** [comment]
Is there plans for this to be implemented in a near future? What could be done to help speedup this process? 

I was under the impression that WPS 2.0 was already handled by PyWPS because the WPS *GetCapabilities* response returns 
```xml
<ows:ServiceTypeVersion>1.0.0</ows:ServiceTypeVersion>
<ows:ServiceTypeVersion>2.0.0</ows:ServiceTypeVersion>
```

Recently, the WPS 2.0 test suite was released, and running a server with the latest PyWPS implementation immediately fails it
(see https://github.com/opengeospatial/ets-wps20/issues/40).

The code should be updated either by removing this incorrect reporting of 2.0 support, or by adding the missing implementations.
Beside the "dismiss" operation, is the rest of the process description and execution relatively equivalent to what PyWPS offers? 
Would it be possible to start with a subset of features of WPS 2.0 until "dismiss" is implemented?
Or, is WPS 2.0 completely skipped in favour of new *OGC API - Processes*?
